src/main/scala/SerialAdapter.scala

```diff
@@ -341,3 +382,117 @@ class SerialRAM(
+  val clientParams = managerEdge.master // The clientParams are the chip-side managerParams
+  val adapter = LazyModule(new SerialAdapter)
+  val serdesser = LazyModule(new TLSerdesser(
+    w,
+    clientParams,
+    managerParams
+  ))
+
+  // connect the serial adapter to serdes manager
+  serdesser.managerNode := TLBuffer() := adapter.node
+
+  val memClkRstDomain = LazyModule(new ClockSinkDomain(name=Some("mem-over-serialtl-domain")))
+  val memClkRstSource = ClockSourceNode(Seq(ClockSourceParameters()))
+  memClkRstDomain.clockNode := memClkRstSource
+
+  val memCrossing = axiMemOverSerialTLParams.axiClockParams match {
+    case Some(params) => {
+      params.crossingType
+    }
+    case None => {
+      SynchronousCrossing()
+    }
+  }
+
+  val memXbar = memClkRstDomain { TLXbar() }
+
+  (memClkRstDomain.crossIn(memXbar)(ValName("MemPortCrossing")))(memCrossing) := serdesser.clientNode
+
+  // TODO: Currently only supports single-channel memory
+  val memPortParamsOpt = Some(MemoryPortParams(memParams, 1))
+  val portName = "axi4"
+  val device = new MemoryDevice
+  val idBits = memPortParamsOpt.map(_.master.idBits).getOrElse(1)
+  val memBusParams = p(MemoryBusKey)
+
+  val memNode = memClkRstDomain {
+    AXI4SlaveNode(memPortParamsOpt.map({ case MemoryPortParams(memPortParams, nMemoryChannels) =>
+      Seq.tabulate(nMemoryChannels) { channel =>
+        val base = AddressSet.misaligned(memPortParams.base, memPortParams.size)
+        val filter = AddressSet(channel * memBusParams.blockBytes, ~((nMemoryChannels-1) * memBusParams.blockBytes))
+
+        AXI4SlavePortParameters(
+          slaves = Seq(AXI4SlaveParameters(
+            address       = base.flatMap(_.intersect(filter)),
+            resources     = device.reg,
+            regionType    = RegionType.UNCACHED, // cacheable
+            executable    = true,
+            supportsWrite = TransferSizes(1, memBusParams.blockBytes),
+            supportsRead  = TransferSizes(1, memBusParams.blockBytes),
+            interleavedId = Some(0))), // slave does not interleave read responses
+          beatBytes = memPortParams.beatBytes)
+      }
+    }).toList.flatten)
+  }
+
+  val memPort = memClkRstDomain {
+    // connect axi mem to axi widgets to mem xbar
+    (memNode
+      :*= AXI4UserYanker()
+      :*= AXI4IdIndexer(idBits)
+      :*= TLToAXI4()
+      :*= TLWidthWidget(memBusParams.beatBytes)
+      :*= memXbar)
+
+    InModuleBody { memNode.makeIOs() }
+  }
+
+  val mem_axi4 = InModuleBody {
+    val ports: Seq[ClockedAndResetIO[AXI4Bundle]] = memPort.zipWithIndex.map({ case (m, i) =>
+      val port = IO(new ClockedAndResetIO(DataMirror.internal.chiselTypeClone[AXI4Bundle](m))).suggestName(s"axi4_mem_${i}")
+      port.bits <> m
+      port.clock := memClkRstSource.out.head._1.clock
+      port.reset := memClkRstSource.out.head._1.reset
+      port
+    })
+    ports
+  }
+
+  lazy val module = new LazyModuleImp(this) {
+    val io = IO(new Bundle {
```

Comment:
Make this a LazyRawModuleImp if the implicit clock/reset aren't used

Reply:
This module still uses the implicit clock/reset for the `SerialAdapter`/`TLSerdesser`/`TLBuffer`